### PR TITLE
prowgen: generate stricter `branch:` stanzas for presubmits

### PR DESCRIPTION
**Pull request by petr-muller**
- prowgen: generate stricter `branch:` stanzas for presubmits
- prowgen: regenerate pj-rehearse test data
- pj-rehearse: handle presubmits with multiple branches

Reintroduce https://github.com/openshift/ci-tools/pull/2136 that needed to be
reverted because pj-rehearse was unable to handle the new presubmit shape. I
regenerated the pj-reharse test data to reflect the change in Prowgen and
therefore reproduce the pj-rehearse problem. Then I fixed pj-rehearse by
processing all `Brancher` items in a presubmit and extracting a "simple name"
from what is potentially a regex. If at least one branch name is successfully
extracted, it is used for rehearsal.

Resolves: [DPTP-2414](https://issues.redhat.com/browse/DPTP-2414)
Doc update: https://github.com/openshift/ci-tools/pull/2199
o/release update: https://github.com/openshift/release/pull/20738

### CODE REVIEW

**Comment by petr-muller** [comment]
/hold
Needs regenerate/merge https://github.com/openshift/release/pull/20738 after merge

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**
